src/wrapper/jnienv.rs

```diff
@@ -274,22 +274,22 @@ impl<'a> JNIEnv<'a> {
 
     /// Create a new instance of a direct java.nio.ByteBuffer.
     pub fn new_direct_byte_buffer(&self, data: &mut [u8]) -> Result<JByteBuffer> {
-        let obj = unsafe {
-            jni_unchecked!(
-                self.internal,
-                NewDirectByteBuffer,
-                data.as_mut_ptr() as *mut c_void,
-                data.len() as jlong
-            )
-        };
+        let obj: JObject = jni_non_null_call!(
+            self.internal,
+            NewDirectByteBuffer,
+            data.as_mut_ptr() as *mut c_void,
+            data.len() as jlong
+        );
         Ok(JByteBuffer::from(obj))
     }
 
     /// Returns the starting address of the memory of the direct
     /// java.nio.ByteBuffer.
     pub fn get_direct_buffer_address(&self, buf: JByteBuffer) -> Result<&mut [u8]> {
+        non_null!(buf, "get_direct_buffer_address argument");
         let ptr: *mut c_void =
             unsafe { jni_unchecked!(self.internal, GetDirectBufferAddress, buf.into_inner()) };
+        non_null!(ptr, "get_direct_buffer_address return value");
```

Comment:
I wonder if we shall have two macros:
- `non_null` — when `NULL` is strictly not allowed;
- and `null_to_err` — when it is allowed, but must be converted to an `Err` (as in this and many other cases)

They _may_ or may not have the same implementation, but document the intent clearer. The last macro `null_to_err` is also applicable to cases where we check preconditions: the null argument is "not allowed", but `Err` is a natural way to report that in Rust, if I understand correctly.

@vitvakatu , @DarkEld3r , WDYT?

Reply:
What do you mean by `allowed, but must be converted to an Err`? I think the only valid behavior would be returning `Err` immediately in case of null.

Reply:
> allowed, but must be converted to an Err

By that I mean a situation when a JNI method is specified to return `null` to indicate an error. By NULL being strictly not allowed I mean a method that is specified to **never** return `null` — I doubt there are many of them.

How to treat preconditions — that’s a separate concern. Some arguments are specified to be strictly non-NULL, but I am not sure if we shall panic in that case. 

Finally, any _Java_ method may return `null` (and any field can be `null`), and it is not an indication that it has failed, please see #148 for a more detailed discussion.